### ISSUE TITLE
New callback syntax

### ISSUE THREAD

**Issue by eric-burel** [VulcanJS/Vulcan#2535]
Since we move to NPM, it's a good time to rethink a few things:

Proposal:
```
Callbacks = {}
// Equivalent to a `lodash.set(Callbacks, "whatever.foobar", { callbacks: [ ], add: () => { TODO }, run: () {TODO})
// Init the Callbacks object with add and run method, handle nesting
registerCallback('whatever.foobar')
// Store a callback in Callbacks.whatever.foobar.callbacks
Callbacks.whatever.foobar.add()
// Run callback in Callbacks.whatever.foobar.callbacks
Callbacks.whatever.foobar.run()
```
This way it's easy to create complex callbacks structure, using a string, but also avoid mistakes in naming when running callbacks.

**Comment by SachaG** [comment]
Why not. I think the important thing is getting CRUD callbacks right and I think the current API works pretty well. For "non-standard" callbacks I don't have a strong opinion but we should think about how to make them work well with typescript and VS Code.

**Comment by eric-burel** [comment]
Yeah this change wouldn't affect how they work with TS and VS Code, it will behave the same as with a string.
Not sure either what's best, I'll leave the proposal here while we think about it. Its main advantage is having a syntax similar to CRUD operations.
